src/Yaapii.Zip/Unprotected.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Yaapii.Atoms;
+using Yaapii.Atoms.Error;
+
+namespace Yaapii.Zip
+{
+    /// <summary>
+    /// Takes a ZipArchive and unprotects each file
+    /// </summary>
+    public sealed class Unprotected : IInput
+    {
+        private readonly string password;
+        private readonly IInput origin;
+
+        /// <summary>
+        /// Takes a ZipArchive and unprotects each file
+        /// </summary>
+        public Unprotected(string password, IInput origin)
+        {
+            this.password = password;
+            this.origin = origin;
+        }
+
+        public Stream Stream()
+        {
+            new FailWhen(
+                () => !new IsZipArchive(this.origin).Value(),
+                new ArgumentException("Can not unprotect zip, because the input is not a ZipArchive")
+            ).Go();
+            var result = new List<KeyValuePair<string, IInput>>();
+            foreach (var file in new ZipFiles(origin, true))
+            {
+                new FailWhen(
+                    () => !new HasPassword(
```

Comment:
Is this Validation necessary? I think it would be easier to use, if this class just unprotects all protected files, but leaves all other files as they are